.github/workflows/update-docker-image.yml

```diff
@@ -0,0 +1,37 @@
+name: update_docker_image
+on:
+  workflow_run:
+    workflows: ["pycram_docker_ci"]
+    types:
+      - completed
+  workflow_dispatch: # manual run
+    inputs:
```

Comment:
The Dockerfile only needs to be updated if the requirements change. We could also automate this: https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#running-your-workflow-only-when-a-push-affects-specific-files